beacon-chain/state/state-native/getters_block.go

```diff
@@ -45,39 +47,60 @@ func (b *BeaconState) latestBlockHeaderVal() *ethpb.BeaconBlockHeader {
 
 // BlockRoots kept track of in the beacon state.
 func (b *BeaconState) BlockRoots() [][]byte {
-	if b.blockRoots == nil {
-		return nil
-	}
-
 	b.lock.RLock()
 	defer b.lock.RUnlock()
 
-	return b.blockRoots.Slice()
+	roots := b.blockRootsVal()
+	if roots == nil {
+		return nil
+	}
+	if features.Get().EnableExperimentalState {
+		rootsSlice := make([][]byte, len(roots))
+		for i, r := range roots {
+			rootsSlice[i] = r[:]
```

Comment:
Wow. I guess all core paths use the private `blockRootsVal()` function and this one is used by non-critical things. Unit tests would fail if they set the flag - should we do that? Turn on the feature in tests by default?